crates/bevy_asset/src/event.rs

```diff
@@ -45,6 +45,24 @@ impl<A: Asset> From<&AssetLoadFailedEvent<A>> for UntypedAssetLoadFailedEvent {
 }
 
 /// [`Message`]s that occur for a specific loaded [`Asset`], such as "value changed" events and "dependency" events.
+/// Events concerning a specific [`Asset`] value.
+///
+/// Each variant carries the [`AssetId`] of the asset affected by the event. These
+/// messages are emitted by the asset system to notify other systems about
+/// lifecycle changes for individual assets:
+///
+/// - `Added`: a new asset value was inserted into the asset storage.
```

Comment:
IMO these are best left to the individual variants. We should improve the docs there as needed, and avoid duplicating this information.